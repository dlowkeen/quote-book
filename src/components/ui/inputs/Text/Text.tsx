import React from 'react';

import * as styles from '../inputs.css';

interface ITextProps {
  label: string;
}

const Text: React.FunctionComponent<ITextProps> = ({ label }) => (
  <div className={`${styles.formElement} ${styles.formElementText}`}>
    <label>{label}</label>
    <input type='text' />
  </div>
);

export default Text;
