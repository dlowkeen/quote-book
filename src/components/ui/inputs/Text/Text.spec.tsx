import React from 'react';
import renderer from 'react-test-renderer';

import Text from './Text';

describe('The Text component', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<Text label='i have something i need to tell you' />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
