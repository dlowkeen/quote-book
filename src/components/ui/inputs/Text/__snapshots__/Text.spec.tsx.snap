// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text component renders as expected 1`] = `
<div
  className="inputs__form-element__VT4Ts inputs__form-element-text__13VAX"
>
  <label>
    i have something i need to tell you
  </label>
  <input
    type="text"
  />
</div>
`;
